pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -336,6 +365,95 @@ func (o *EditOptions) Run() error {
 	}
 }
 
+func (o *EditOptions) applyEditPatch(oringalInfos, infos []*resource.Info) error {
+	errors := []error{}
+	for ix := range infos {
+		updatedInfo := infos[ix]
+
+		originalJS, err := encodeToJson(o.Encoder, oringalInfos[ix].Object)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		editedJS, err := encodeToJson(o.Encoder, updatedInfo.Object)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		if reflect.DeepEqual(originalJS, editedJS) {
+			cmdutil.PrintSuccess(o.Mapper, false, o.Out, updatedInfo.Mapping.Resource, updatedInfo.Name, false, "skipped")
+		} else {
+			err := o.annotationPatch(updatedInfo)
+			if err != nil {
+				errors = append(errors, err)
+			}
+			cmdutil.PrintSuccess(o.Mapper, false, o.Out, updatedInfo.Mapping.Resource, updatedInfo.Name, false, "edited")
+		}
+	}
+	if len(errors) > 0 {
+		return utilerrors.NewAggregate(errors)
+	}
+
+	return nil
+}
+
+func (o *EditOptions) annotationPatch(update *resource.Info) error {
+	patch, _, err := GetApplyPatch(update, o.Encoder, false)
+	if err != nil {
+		return err
+	}
+	mapping := update.ResourceMapping()
+	client, err := o.f.UnstructuredClientForMapping(mapping)
+	if err != nil {
+		return err
+	}
+	helper := resource.NewHelper(client, mapping)
+	_, err = helper.Patch(o.CmdNamespace, update.Name, types.StrategicMergePatchType, patch)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetApplyPatch(info *resource.Info, codec runtime.Encoder, isLocal bool) ([]byte, []byte, error) {
+	if info == nil {
+		return []byte("{}"), []byte(""), nil
+	}
+	var obj runtime.Object
+	if isLocal {
+		obj = info.VersionedObject
+	} else {
+		obj = info.Object
+	}
+	data, err := runtime.Encode(codec, obj)
+	if err != nil {
+		return nil, data, err
+	}
+	if len(data) == 0 {
+		return []byte("{}"), []byte(""), nil
+	}
+	objMap := map[string]map[string]map[string]string{}
```

Comment:
this is an awkward way to build a patch, and assumes annotations will always be represented the same way for all objects. The correct way is to:
1. serialize JSON
2. set the annotation using a meta.Accessor
3. serialize JSON again
4. compute the merge patch